system/Helpers/array_helper.php

```diff
@@ -218,3 +218,49 @@ function array_flatten_with_dots(iterable $array, string $id = ''): array
         return $flattened;
     }
 }
+
+if (! function_exists('array_group_by')) {
+    /**
+     * Groups all rows by their index values. Result's depth equals number of indexes
+     *
+     * @param array $array        Data array (i.e. from query result)
+     * @param array $indexes      Indexes to group by. Dot syntax used. Returns $array if empty
+     * @param bool  $includeEmpty If true, null and '' are also added as valid keys to group
+     *
+     * @return array Result array where rows are grouped together by indexes values.
+     */
+    function array_group_by(array $array, array $indexes, bool $includeEmpty = false): array
+    {
+        if (empty($indexes)) {
+            return $array;
+        }
+
+        $result = [];
+
+        foreach ($array as $row) {
+            $currentLevel = &$result;
+            $valid        = true;
+
+            foreach ($indexes as $index) {
+                $value = dot_array_search($index, $row);
+
+                if (! $includeEmpty && empty($value)) {
```

Comment:
Please use strict instead of `empty`.

Reply:
I replaced `empty($value)` with `($value === null || $value === '')`. Furthermore the value is not valid in case it is an array or an object (only numbers and strings allowed)